File: Lambda/Intent/app/src/main/java/com/example/intent/MainActivity.java
package com.example.intent;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    public void changeActivity(View view){
        Toast.makeText(this,"button click !",Toast.LENGTH_SHORT).show();
        Log.i("test","pressed!");

    }

    View.OnClickListener l = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            Toast.makeText(MainActivity.this, "onClickListener function!", Toast.LENGTH_SHORT).show();
            Log.i("test", "onClickListener function!");
            Intent intent = new Intent(getApplicationContext(), Activity2.class);
            intent.putExtra("data", "hello");
            startActivity(intent);

        }
    };

    View.OnClickListener requestlistenser = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            Toast.makeText(MainActivity.this, "request function!", Toast.LENGTH_SHORT).show();
            Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
            intent.setType("image/*");
            if (intent.resolveActivity(getPackageManager()) != null){
                startActivityForResult(intent,2);
            }

        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.button2);
        button.setOnClickListener(l);

        Button requesButton = findViewById(R.id.request);
        requesButton.setOnClickListener(requestlistenser);


    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode==2 && resultCode== Activity.RESULT_OK){
            Toast.makeText(this, "Receving image",Toast.LENGTH_SHORT).show();
        }
    }
}

File: Lambda/MyApplication/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        showName.setOnClickListener{
            Toast.makeText(this, "Hello Kolin", Toast.LENGTH_SHORT).show();
            val intent = Intent(this, Activity3::class.java);
            intent.putExtra("Data", "Derek");
            startActivity(intent);
        }

        requestPhoto.setOnClickListener {
            val intent = Intent(Intent.ACTION_GET_CONTENT);
            intent.type="image/*"
            startActivityForResult(intent,1);

        }




    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode==1 && resultCode== Activity.RESULT_OK){
            Toast.makeText(this, "image received! ", Toast.LENGTH_SHORT).show();
        }
    }
}
